vscode/src/testExplorer.ts

```diff
@@ -0,0 +1,267 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// This file uses the VS Code Test Explorer API (https://code.visualstudio.com/docs/editor/testing)
+
+import * as vscode from "vscode";
+import { loadProject } from "./projectSystem";
+import {
+  getCompilerWorker,
+  ICompilerWorker,
+  IProjectConfig,
+  log,
+  ProgramConfig,
+  QscEventTarget,
+} from "qsharp-lang";
+import { getActiveQSharpDocumentUri } from "./programConfig";
+import { IProgramConfig } from "../../npm/qsharp/lib/web/qsc_wasm";
+import { getTarget } from "./config";
+import { toVsCodeRange } from "./common";
+
+function localGetCompilerWorker(
+  context: vscode.ExtensionContext,
+): ICompilerWorker {
+  const compilerWorkerScriptPath = vscode.Uri.joinPath(
+    context.extensionUri,
+    "./out/compilerWorker.js",
+  ).toString();
+  const worker = getCompilerWorker(compilerWorkerScriptPath);
+  return worker;
+}
+
+async function getProgramConfig(): Promise<IProgramConfig | null> {
+  if (!vscode.workspace.workspaceFolders) {
+    log.info("No workspace detected; not starting test explorer");
+    return null;
+  }
+
+  const docUri = getActiveQSharpDocumentUri();
+  if (!docUri) {
+    log.info("No active document detected; not starting test explorer");
+    return null;
+  }
+
+  const projectConfig: IProjectConfig = await loadProject(docUri);
+  if (!projectConfig) {
+    log.info("No project detected; not starting test explorer");
+    return null;
+  }
+
+  return {
+    profile: getTarget(),
+    ...projectConfig,
+  };
+}
+
+/**
+ * Constructs the handler to pass to the `TestController` that refreshes the discovered tests.
+ * if `shouldDeleteOldTests` is `true`, then clear out previously discovered tests. If `false`, add new tests but don't dissolve old ones.
+ *
+ */
+function mkRefreshHandler(
+  ctrl: vscode.TestController,
+  context: vscode.ExtensionContext,
+  shouldDeleteOldTests: boolean = true,
```

Comment:
This function is only called twice, once without `shouldDeleteOldTests` (so it defaults to `true`), and once passed `true` explicitly, so is there any value in having this parameter?

Reply:
This is a remnant of refactoring. Removed.